### DISCUSSION TITLE
Installing Pybricks firmware on spike prime from windows / Mac not working

### DISCUSSION THREAD

**Discussion by SVD81Tech** [pybricks/support#1900] [Q&A]
Hello Team,

i am following the instructions to install the firmware on my spike prime 3. 
unfortunately after shutting down, when i press the bluetooth and connected it to USB and try to install on both windows or Mac my device is not detected. and i have no clue how get the drivers installed. 
some help will be really appreciated. 

Thanks,

Sushil

**Comment by dlech** [comment]
Bad USB cables seem to be a common issue that causes the hub to not be detected. 

Also, can you confirm that the Bluetooth light on the hub is flashing red/green/blue/off?

**Reply by SVD81Tech** [reply]
thanks for the lightening response. 
yes the blue tooth light blinks red / green / blue 
i changed couple of wires. Also the green charging light switches on when the wire is connected to the  laptop / mac. And same wire is used for charging the hub. 
Also when i connected the hub to my mac i used Ankler USBC to USB female connector and then USB male to the hub micro USB . not sure if that is an issue. bcas when i go to Mac / system info / hardware / USB i see Ankler device but it doesnt show me spike hub

**Reply by dlech** [reply]
Some USB cables can only do charging and not data, so a different cable might help.

You could also try taking the battery out and starting over in case the hub somehow got in a bad state.

**Reply by SVD81Tech** [reply]
Thanks a ton !!!
wire was a culprit. bought new wire and it worked. 
thank you again !!! you guys rock!!!

**Reply by laurensvalk** [reply]
Have fun coding!
